File: readability/readability.c
#include <ctype.h>
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <math.h>

// Declare count letters function
int count_letters(string text);

// Declare count words function
int count_words(string text);

// Declare count sentences function
int count_sentences(string text);

int main(void)
{
// Get text from user
    string text = get_string("Text: ");

// Convert sentences to float
    float s = count_sentences(text);

// Convert words to float
    float w = count_words(text);

// Convert letters to float
    float l = count_letters(text);

// Calculate readability with formula
    float index = 0.0588 * (l / w * 100) - 0.296 * (s / w * 100) - 15.8;

// Print the result
    if (index >= 1 && index < 16)
    {
        printf("Grade %i\n", (int) round(index));
    }

    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }

    else if (index >= 16)
    {
        printf("Grade 16+\n");
    }
}

// Count letters function
int count_letters(string text)
{
    float l = 0;

    for (int i = 0; text[i] != '\0'; i++)
    {
        if (isupper(text[i]) || islower(text[i]))
        {
            l++;
        }
    }
    return l;
}

// Count words function
int count_words(string text)
{
    float w = 1;

    for (int i = 0; text[i] != '\0'; i++)
    {
        if (text[i] == ' ')
        {
            w++;
        }
    }
    return w;
}

// Count sentences function
int count_sentences(string text)
{
    float s = 0;

    for (int i = 0; text[i] != '\0'; i++)
    {

        if (text[i] == '.' || text[i] == '!' || text[i] == '?')
        {
            s++;
        }
    }
    return s;
}

File: sentimental-readability/readability.py
from cs50 import get_string


def main():

    # Get text from user
    text = get_string("Text: ")

    # Convert sentences to float
    s = float(count_sentences(text))

    # Convert words to float
    w = float(count_words(text))

    # Convert letters to float
    l = float(count_letters(text))

    # Calculate readability with formula
    index = float(0.0588 * ((l / w) * 100) - 0.296 * ((s / w) * 100) - 15.8)

    # Print the result
    if index >= 1 and index < 16:
        print(f"Grade {index}")

    elif index < 1:
        print("Before Grade 1")

    elif index >= 16:
        print("Grade 16+")


# Count letters function
def count_letters(text):
    l = 0
    for i in range(len(text)):
        if text[i] != ' ' and text[i] != '.' and text[i] != '!' and text[i] != '?':
            l += 1
    return l


# Count words function
def count_words(text):
    w = 1
    for i in range(len(text)):
        if text[i] == ' ':
            w += 1
    return w


# Count sentences function
def count_sentences(text):
    s = 0
    for i in range(len(text)):
        if text[i] == '.' or text[i] == '!' or text[i] == '?':
            s += 1
    return s


main()

